osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1309 @@
+  char buf[max];
+
+  switch (logType) {
+  case IPMI_LOG_SEVERE:
+  case IPMI_LOG_FATAL:
+  case IPMI_LOG_ERR_INFO:
+    vsnprintf(buf, max, format, ap);
+    LOG(ERROR) << buf;
+
+  default:
+    // Silence all other log levels
+    break;
+  }
+}
+
+/// Converts bytes from first to length len to CIDR format in std::string.
+static inline std::string toCIDR(const unsigned char* first,
+                                 const unsigned int len) {
+  if (len != 5) {
+    LOG(ERROR) << "Unexpected return data length for CIDR PARM data of " << len
+               << "; expected 5";
+    return "";
+  }
+
+  std::stringstream cidr;
+  for (unsigned int i = 1; i < len; i++) {
+    cidr << static_cast<int>(first[i]);
+    if (i != len - 1) {
+      cidr << ".";
+    }
+  }
+
+  return cidr.str();
+}
+
+/// Converts bytes from first to length len to MAC format in std::string.
+static inline std::string toMAC(const unsigned char* first,
+                                const unsigned int len) {
+  if (len != 7) {
+    LOG(ERROR) << "Unexpected return data length for MAC address PARM data of "
+               << len << "; expected 7";
+    return "";
+  }
+
+  std::stringstream mac;
+  for (unsigned int i = 1; i < len; i++) {
+    mac << std::hex << static_cast<int>(first[i]);
+    if (i != len - 1) {
+      mac << ":";
+    }
+  }
+
+  return mac.str();
+}
+
+/// Converts bytes to stringified IP address source.
+static inline std::string toAddressSrc(const unsigned char* val,
+                                       const unsigned int len) {
+  if (len != 2) {
+    LOG(WARNING)
+        << "Expected data length of 2 for IPMI_LANPARM_IP_ADDRESS_SRC, but got "
+        << std::to_string(len);
+    return "";
+  }
+
+  switch (val[1] & 7) {
+  case IPMI_LANPARM_IP_ADDR_SRC_STATIC:
+    return "static";
+
+  case IPMI_LANPARM_IP_ADDR_SRC_DHCP:
+    return "dhcp";
+
+  case IPMI_LANPARM_IP_ADDR_SRC_BIOS:
+    return "bios";
+
+  case IPMI_LANPARM_IP_ADDR_SRC_OTHER:
+    return "other";
+
+  default:
+    return "unknown";
+  }
+}
+
+/// Converts bytes to stringified set progress state.
+static inline std::string toSetInProgress(const unsigned char* val,
+                                          const unsigned int len) {
+  if (len != 2) {
+    LOG(WARNING) << "Expected data length of 2 for "
+                    "IPMI_LANPARM_SET_IN_PROGRESS, but got "
+                 << std::to_string(len);
+
+    return "";
+  }
+
+  switch (val[1] & 3) {
+  case 0:
+    return "set complete";
+  case 1:
+    return "set in progress";
+  case 2:
+    return "commit write";
+  default:
+    return "unknown";
+  }
+}
+
+/* @brief Handles the unlocking of ipmi_lanparm_t*.
+ *
+ * @param lp ipmi_lanpar_t* to unlock
+ *
+ * Debug logs if error is encountered, since depending on system this can
+ * error out and be inconsequential.
+ */
+static inline void unlockLANPARM(ipmi_lanparm_t* lp) {
+  int rv =
+      ipmi_lan_clear_lock(lp,
+                          NULL,
```

Comment:
So i actually had some issues using `nullptr` in openipmi api calls instead of `NULL` (not really sure why, didn't really look into it, but using `NULL` in place got everything working as expected`).  I'll retest after I get the other changes in.